edb/server/protocol/protocol.pyx

```diff
@@ -541,6 +544,17 @@ cdef class HttpProtocol:
                     await edgeql_ext.handle_request(
                         request, response, db, args, self.server
                     )
+                elif extname == 'auth':
+                    netloc = (
+                        f"{request.url.host.decode()}:{request.url.port}"
+                            if request.url.port
+                            else request.url.host.decode()
+                    )
+                    extension_base_path = f"{request.url.schema.decode()}://" \
+                                          f"{netloc}/db/{dbname}/ext/auth"
```

Comment:
I think there is also the `X-Forwarded-*` [headers family](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For) to consider here.

Reply:
Would that be [`X-Forwarded-Host`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host) in this case?

Reply:
Yup, as well as [`X-Forwarded-Proto`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto) for the original protocol.

Sorry about the previous misleading link.

Reply:
So, I'm thinking something like `def _get_netloc(request) -> str:` that does a bunch of lookups with fallbacks to the `request.url`, is that roughly the suggestion here?

Reply:
Yeah, that'll do! 🙏

Reply:
Alright, I went down a little rabbit hole about how to parse this and the related standardized `Forwarded` header, and I'm not sure we can just blanket trust these headers. It seems like servers tend to make trusting the proxy a configurable thing with whitelist IP ranges, and often need some tweaking.

Ultimately, if what we're trying to support here is custom domains or reverse proxies, I wonder if we should make this configurable? In that case, instead of trying to parse the headers ourselves, we have the user tell us what the callback_uri base ought to be and if that configuration value is unset (once we support that...) we fall back to the `request.url`. Thoughts?

Reply:
After some discussion, we decided that we'll need to address multiple cases where the `request.url` is not canonical, and each case will likely be addressed slightly differently. Will address as part of ongoing work on getting this ready for production.